packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -1,175 +1,233 @@
-    );
-  }
-
-  private _onControlledChange = (ev: React.FormEvent<HTMLElement>, checked: boolean): void => {
-    if (this.state.indeterminateState) {
-      this.setState({ indeterminateState: false });
-    } else {
-      this.setState({ checkedState: checked! });
-    }
+const IndeterminateControlledCheckbox: React.FunctionComponent = () => {
+  const [indeterminate, setIndeterminate] = React.useState(true);
+  const [checked, setChecked] = React.useState(false);
+  const onChange = (ev: React.FormEvent<HTMLElement>, newChecked: boolean): void => {
+    // On first change, clear the indeterminate state and don't modify the checked state
+    indeterminate ? setIndeterminate(false) : setChecked(!!newChecked);
   };
-}
+
+  return <Checkbox indeterminate={indeterminate} checked={checked} onChange={onChange} componentRef={checkboxRef} />;
+};
 
 describe('Checkbox', () => {
-  it('renders Checkbox correctly', () => {
-    const component = renderer.create(<Checkbox label="Standard checkbox" />);
-    const tree = component.toJSON();
+  let renderedComponent: renderer.ReactTestRenderer | undefined;
+  let component: ReactWrapper | undefined;
+
+  afterEach(() => {
+    checkbox = undefined;
+    if (renderedComponent) {
+      renderedComponent.unmount();
+      renderedComponent = undefined;
+    }
+    if (component) {
+      component.unmount();
+      component = undefined;
+    }
+  });
+
+  it('renders unchecked correctly', () => {
+    renderedComponent = renderer.create(<Checkbox label="Standard checkbox" />);
+    const tree = renderedComponent.toJSON();
     expect(tree).toMatchSnapshot();
   });
-  it('renders Checkbox with id correctly', () => {
-    const component = mount(<Checkbox label="Standard checkbox" ariaDescribedBy={'descriptionID'} id="my-checkbox" />);
+
+  it('renders checked correctly', () => {
+    renderedComponent = renderer.create(<Checkbox label="Standard checkbox" checked />);
+    const tree = renderedComponent.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders indeterminate correctly', () => {
+    renderedComponent = renderer.create(<Checkbox label="Standard checkbox" indeterminate />);
+    const tree = renderedComponent.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('respects id prop', () => {
+    component = mount(<Checkbox label="Standard checkbox" ariaDescribedBy="descriptionID" id="my-checkbox" />);
     expect(component.find('input').prop('id')).toEqual('my-checkbox');
   });
-  it('renders Checkbox with Uncontrolled indeterminate correctly', () => {
-    const component = mount(<Checkbox label="Uncontrolled Indeterminate checkbox" defaultIndeterminate />);
-    expect(component.find(CheckboxBase).state('isIndeterminate')).toEqual(true);
+
+  it('defaults to unchecked non-indeterminate', () => {
+    component = mount(<Checkbox componentRef={checkboxRef} />);
+
+    const input = component.find('input');
+    expect(input.prop('checked')).toBe(false);
```

Comment:
- Add tests for `checked`, `defaultChecked`, and `onChange` in addition to indeterminate stuff
- Verify both the item props and the component state (to ensure they're in sync)